skbio/__init__.py

```diff
@@ -79,3 +77,8 @@
     __doc__ = title + art
 else:
     __doc__ = title + art + __doc__
+
+test = TestRunner(__file__).test
+
+if __name__ == '__main__':
+    test()
```

Comment:
I think exiting here with `sys.exit` will take care of the exit code issue. For example:

```python
sys.exit(test())
```

And then `test()` would need to return the exit code from invoking the tests. checklist.py takes a similar approach.